src/NServiceBus.NHibernate/Outbox/OutboxPersister.cs

```diff
@@ -101,22 +101,13 @@ await tx.CommitAsync()
             }
         }
 
-        public async Task<OutboxTransaction> BeginTransaction(ContextBag context)
+        public Task<OutboxTransaction> BeginTransaction(ContextBag context)
         {
             //Provided by Get
             var endpointQualifiedMessageId = context.Get<string>(EndpointQualifiedMessageIdContextKey);
-
             var result = outboxTransactionFactory();
-            try
-            {
-                await result.Begin(endpointQualifiedMessageId).ConfigureAwait(false);
-                return result;
-            }
-            catch (Exception e)
-            {
-                result.Dispose();
-                throw new Exception("Error while opening outbox transaction", e);
-            }
+            result.Prepare();
+            return result.Begin(endpointQualifiedMessageId);
```

Comment:
Is the try catch no longer needed?

Shouldn't this be restored back to

            try
            {
                result.Prepare();
                await result.Begin(endpointQualifiedMessageId).ConfigureAwait(false);
                return result;
            }
            catch (Exception e)
            {
                result.Dispose();
                throw new Exception("Error while opening outbox transaction", e);
            }

Reply:
Never mind. The outer layers always dispose of the outbox transaction https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveToPhysicalMessageConnector.cs#L44